org.alloytools.alloy.application/src/main/java/templates/TestTemplateFile.vm

```diff
@@ -70,159 +71,117 @@ class State(ABC):
-                logging.log(SVLOG, self.name + "'s parent " + self.parent.name + " does not have valid value.")
-                return False
-
-        # Multiplicity
-        if (self.upper and len(value) > self.upper) or self.lower > len(value):
-            logging.log(SVLOG, self.name + " violates the multiplicity rule.")
-            return False
+    def __repr__(self):
+        return f"{self.sigName} - {self.name} : {self.elements}" if self.elements else f"{self.sigName} - {self.name} : Empty"
 
-        return True
+    def __contains__(self, other):
+        if isinstance(other, MetaSignature):
+            return other.possible_objects().issubset(self.elements)
+        if isinstance(other, Signature):
+            return other.elements.issubset(self.elements)
+        if isinstance(other, str):
+            return other in self.elements
+        return False
 
-    def get_value(self):
-        if self.subsigs:
-            subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-            return subset.union(self._value)
-        return self._value
-
-    # Returns None if the value intended to be set is invalid
-    def set_value(self, value):
-        if self.check_validity(value):
-            subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-            self._value = value - subset
-            return self._value
-        else:
-            return None
-
-    def set_value_no_constraints(self, value):
-        subset = set([v for subsig in self.subsigs for v in subsig.get_value()])
-        self._value = value - subset
-        return self._value
 
-    def __repr__(self):
-        return f"{self.name} {self.get_value()}"
+    def set_failed(self, new_elements) -> bool:
+        if(type(self).is_valid(new_elements)):
+            self.elements = new_elements
+            return False
+        return True
+
 
 # --- Signatures ---
 
-#foreach($signature in $signatures)
-${signature.Name} = Signature("${signature.Name}", "${signature.Multiplicity}", ${signature.Cardinality}, ${signature.IsSubset}, ${signature.Parents}, ${signature.IsSubsig}, ${signature.Parent}, ${signature.IsAbstract})
-#end
+#foreach($sig in $signatures)
+    #if(${sig.isSubset()})
+class ${sig.Name}(${sig.getParentsName()}):
+    #elseif(${sig.isSubsig()})
+class ${sig.Name}(${sig.getParentName()}):
+    #else
+class ${sig.Name}(Signature):
+    #end
+    # Signature (static) information
+    sig_objects = {${sig.getObjectNames()}}
+    sigName = "${sig.Name}"
+
+    def __init__(self, name):
+    #if(${sig.isSubset()})
+        # TODO: maybe we don't need to initialize using super() for subsets
+        #foreach($parentName in $sig.getParentNames())
+        ${parentName}.__init__(self, name)
+        #end
+    #elseif(${sig.isSubsig()})
+        ${sig.getParentName()}.__init__(self, name)
+    #else
+        Signature.__init__(self, name)
+    #end
 
+    def possible_objects() -> set:
```

Comment:
"Method must have a first parameter, usually called 'self'"

Reply:
not really

Reply:
Makes sense. Sorry, my linter was complaining about it.